.github/workflows/workflows.yaml

```diff
@@ -0,0 +1,63 @@
+name: Apply new changes to SolEng repositories
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - workflows/**
+
+concurrency:
+      # We do not want to run multiple jobs for single PR.
+      group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
+      cancel-in-progress: true
+
+jobs:
+  workflows:
+    name: Run workflows update
+    runs-on: ubuntu-latest
+    strategy:
+        fail-fast: false
+        matrix:
+          repository:
+            - charm-advanced-routing
+          branch: [main]
+          check_template: [charm-check.yaml.j2]
+          owner: [rgildein]
+          runs_on: ["[[ubuntu-24.04]]"]
+          test_command: ["['tox -e func']"]
+
+    steps:
+      - name: Checkout automation repo
+        uses: actions/checkout@v4
+      - name: Checkout ${{ matrix.repository }}
+
+        uses: actions/checkout@v4
+        with:
+          path: ./.tmp
+          repository: ${{ matrix.owner }}/${{ matrix.repository }}
+          ref: ${{ matrix.branch }}
+      - name: Rendering check workflow template
+        uses: cuchi/jinja2-action@v1.2.2
+        with:
+          template: ./workflows/${{ matrix.check_template }}
+          output_file: ./.tmp/.github/workflows/check.yaml
+          variables: |
+            runs_on=${{ matrix.runs_on }}
+            test_command=${{ matrix.test_command }}
```

Comment:
How will this scale to multiple templates and different templates for different repositories?

Reply:
We define matrix with default template and for those which use separate template we simply include them one by one.

Reply:
Can you show an example in the workflow here?

Reply:
```yaml
        matrix:
          repository:
            - charm-advanced-routing
          branch: [main]
          check_template: [charm-check.yaml.j2]
          owner: [rgildein]
          runs_on: ["[[ubuntu-24.04]]"]
          test_command: ["['tox -e func']"]
        # adding lines like this
        include:
        - repository: ABC
          check_template: ABC-template
          ...
```